frame/support/src/traits/misc.rs

```diff
@@ -407,6 +407,92 @@ where
 	}
 }
 
+/// Defensively checks the minimum between two values.
+pub trait DefensiveMin<T> {
+	/// Defensively checks the minimum between two values
+	/// or if the two values are the same.
+	///
+	/// # Example
+	///
+	/// assert_eq!(10, 10_u32.defensive_min(10_u32));
+	/// ```
+	fn defensive_min(&self, t: T) -> Self;
+	/// Strictly defensively checks the minimum between two values.
+	///
+	/// # Example
+	///
+	/// assert_eq!(10, 10_u32.defensive_strict_min(11_u32));
+	/// ```
+	fn defensive_strict_min(&self, t: T) -> Self;
+}
+
+impl<T, U> DefensiveMin<U> for T
+where
+	T: Clone + sp_std::cmp::PartialOrd<U>,
+	U: Copy + std::fmt::Debug,
```

Comment:
```suggestion
	U: Copy + sp_std::fmt::Debug,
```
Please always use `sp_std` to keep it no-std.

Reply:
Ah, Apologies.
I missed that.